components/badge/badge.vue

```diff
@@ -121,6 +139,24 @@ export default {
       if (this.type === 'ai' && this.kind === 'count') {
         console.error('DtBadge error: type: \'ai\' with kind: \'count\' is an invalid combination.');
       }
+
+      this.validateDecorationPropCombination();
+    },
+
+    validateDecorationPropCombination () {
+      if (this.decoration === 'default') return;
+
+      if (this.kind !== 'label' || this.type !== 'default') {
+        console.error('DtBadge error: decoration prop can only be used with kind: \'label\' and type: \'default\'.');
+      }
+
+      if (this.hasIcons()) {
+        console.error('DtBadge error: decoration prop cannot be used with iconLeft or iconRight.');
+      }
+    },
```

Comment:
`validateDecorationPropCombination() {
  const isDefaultDecoration = this.decoration === 'default';
  const isInvalidDecorationCombination = !isDefaultDecoration && (this.kind !== 'label' || this.type !== 'default');
  const hasIcons = this.hasIcons();

  if (isInvalidDecorationCombination) {
    console.error('DtBadge error: decoration prop can only be used with kind: \'label\' and type: \'default\'.');
  } else if (hasIcons) {
    console.error('DtBadge error: decoration prop cannot be used with iconLeft or iconRight.');
  }
}`

Not necessary to replace but here is a suggestion.